src/web/context/GlobalContext.tsx

```diff
@@ -0,0 +1,252 @@
+// React
+import React, { createContext, ReactNode, useEffect, useReducer } from 'react'
+
+// Jaia
+import { jaiaAPI } from '../jcc/common/JaiaAPI'
+
+export interface GlobalContextType {
+    clientID: string,
+    controllingClientID: string
+    selectedPodElement: SelectedPodElement
+    showHubDetails: boolean
+    hubAccordionStates: HubAccordionStates
+    isRCMode: boolean
+}
+
+export interface SelectedPodElement {
+    type: PodElement
+    id: number
+}
+
+interface HubAccordionStates {
+    quickLook: boolean,
+    commands: boolean,
+    links: boolean
+}
+
+export interface GlobalAction {
+    type: string,
```

Comment:
What if type is an enum, so we can document each action type:

```
export enum GlobalActionType {
    
    /**
     * @param clientID The client id that was saved.
     */
    SAVED_CLIENT_ID = 1,
    TAKE_CONTROL_SUCCESS = 2
    ...
}

export interface GlobalAction {
    type: GlobalActionType,
    clientID?: string,
    hubAccordionName?: string
}
```

Reply:
If we used an enum like the above code, then `@param` shows up when hovering over the type enum:

![image](https://github.com/jaiarobotics/jaiabot/assets/19232537/967413da-3dc6-4d3c-bfd6-c8da7ed3b758)

Reply:
The enum wouldn't have to be integers either, it could be strings.